src/network-analytics/azext_network_analytics/tests/latest/test_network_analytics.py

```diff
@@ -0,0 +1,75 @@
+            self.check('name', '{name}'),                                                                                                     
+            self.check('provisioningState', 'Succeeded')
+        ])
+
+        #wait for 20 second
+        time.sleep(20)
+
+        #List All DP
+        dplist = self.cmd('az network-analytics data-product list').get_output_in_json()
+        assert len(dplist) > 0
+
+        #List DP
+        dplistinrg = self.cmd('az network-analytics data-product list --resource-group {group}').get_output_in_json()
+        assert len(dplistinrg) == 1
+
+        #Get DP
+        self.cmd('az network-analytics data-product show --name {name} --resource-group {group}', checks=[
+            self.check('name', '{name}'),
+            self.check('type', 'microsoft.networkanalytics/dataproducts')
+        ])
+
+        #Add User Role
+        self.cmd('az network-analytics data-product add-user-role --data-product-name {name} --resource-group {group} --data-type-scope " " --principal-id {id} --principal-type user --role reader --role-id " " --user-name " "')
+
+        #wait for 20 second
+        time.sleep(20)
+
+        #List User Role
+        rolelist = self.cmd('az network-analytics data-product list-roles-assignment --data-product-name {name} --resource-group {group}').get_output_in_json()
+        assert rolelist['count'] == 1
+        assert rolelist['roleAssignmentResponse'][0]['principalId'] == user_id
+
+        #Remove User Role
+        self.cmd('az network-analytics data-product remove-user-role --data-product-name {name} --resource-group {group} --data-type-scope " " --principal-id {id} --principal-type user --role reader --role-id " " --user-name " " --role-assignment-id " "')
+
+        #wait for 20 second
+        time.sleep(20)
+
+        #List User Role
+        emptyrolelist = self.cmd('az network-analytics data-product list-roles-assignment --data-product-name {name} --resource-group {group}').get_output_in_json()
+        assert emptyrolelist['count'] == 0
+
+        #Delete DP
+        self.cmd('az network-analytics data-product delete --name {name} --resource-group {group} --yes')
+
+        #wait for 20 second
+        time.sleep(20)
+
+        #List DP
+        dplistinrg = self.cmd('az network-analytics data-product list --resource-group {group}').get_output_in_json()
+        assert len(dplistinrg) == 0
+
+    pass
```

Comment:
We shouldn't use `assert` for testing, our test sdk has its check method. You can refer to other modules' solution.

Reply:
replaced assert with self.assertEquals and self.assertTrue